syntax/expr_binary_test.go

```diff
@@ -17,15 +17,27 @@ func TestRelationCall(t *testing.T) {
 	AssertCodesEvalToSameValue(t, `"val"`, s)
 }
 
-func TestOpsAddArrow(t *testing.T) {
+func TestOpsAddArrowForTuples(t *testing.T) {
 	t.Parallel()
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 2, c: 3, d: 4)`, `(a: 1, b: 2) +> (c: 3, d: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 3, c: 4)`, `(a: 1, b: 2) +> (b: 3, c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 3, c: 4)`, `(a: 1, b: (c: 2)) +> (b: 3, c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 4), c: 4)`, `(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: (b: 1), b: (c: 4))`, `(a: 1, b: (c: 2)) +> (a: (b: 1), b: (c: 4))`)
+}
 
-	AssertCodesEvalToSameValue(t, `(a: 1, b: 2) +> (c: 3, d: 4)`, `(a: 1, b: 2, c: 3, d: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: 2) +> (b: 3, c: 4)`, `(a: 1, b: 3, c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (b: 3, c: 4)`, `(a: 1, b: 3, c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)`, `(a: 1, b: (c: 4), c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (a: (b: 1), b: (c: 4))`, `(a: (b: 1), b: (c: 4))`)
+func TestOpsAddArrowForDicts(t *testing.T) {
```

Comment:
do another test where both `lhs` and `rhs` have no mutual keys. Do the same for tuples.